### PR TITLE
[FEATURE] Allow publish of ELTS releases

### PR DESCRIPTION
**Pull request by andreaskienast**

### CODE REVIEW

File: src/Uploader/AzureUploader.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    public function __construct()
+    {
+        $this->client = ServicesBuilder::getInstance()->createBlobService(getenv('AZURE_CONNECTIONSTRING'));
+        $this->containerName = getenv('AZURE_CONTAINER');
+    }
+
+    public function getUploadStartMessage(): string
+    {
+        $containerName = getenv('AZURE_CONTAINER');
```
Comment (bmack) [inline src/Uploader/AzureUploader.php]: why calling getenv() herer again if it's within $this->containerName? (minor nitpick)

File: src/Uploader/UploaderInterface.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+namespace TYPO3\Darth\Uploader;
+
+/*
+ * This file is part of the TYPO3 project.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+interface UploaderInterface
```
Comment (bmack) [inline src/Uploader/UploaderInterface.php:12]: do we need the interface at all if we don't use it really?

File: src/Command/PublishCommand.php [hunk 3]
```diff
@@ -49,6 +51,13 @@ protected function configure()
                 InputOption::VALUE_OPTIONAL,
                 'Set it to "security" if something special is needed',
                 'bugfix'
+            )
+            ->addOption(
+                'elts',
+                null,
+                InputOption::VALUE_OPTIONAL,
```
Comment (bmack) [inline src/Command/PublishCommand.php]: I guess we don't need a value at all for this option then `--elts` should be sufficient. (minor nitpick, but no fixing needed)

File: src/Uploader/EltsUploader.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function getUploadStartMessage(): string
+    {
+        $username = getenv('ELTS_REMOTE_USER');
+        $location = getenv('ELTS_REMOTE_LOCATION');
+        return sprintf('Publishing as %s to %s', $username, $location);
+    }
+
+    public function upload(string $file, string $blobName)
```
Comment (bmack) [inline src/Uploader/EltsUploader.php:44]: let's move the PublishService into the Uploader, so the Uploader is not just a shell, but actually doing stuff. Or are there any other reasons for that extraction?

File: src/Uploader/AzureUploader.php (hunk 1 above)
Comment (andreaskienast) [reply src/Uploader/AzureUploader.php]: Good catch!

File: src/Uploader/UploaderInterface.php (hunk 2 above)
Comment (andreaskienast) [reply src/Uploader/UploaderInterface.php:12]: Well, it's used by AzureUploader and EltsUploader, which allows me to separate the uploaders

File: src/Command/PublishCommand.php (hunk 3 above)
Comment (andreaskienast) [reply src/Command/PublishCommand.php]: I always fail at those argument / option things. I'll check this again.

File: src/Uploader/EltsUploader.php (hunk 4 above)
Comment (andreaskienast) [reply src/Uploader/EltsUploader.php:44]: We can do that

**Review by bmack — APPROVED** [review]

**Pull request merged**